src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -220,9 +220,18 @@ void YARPRobotStatePublisherModule::onRead(yarp::rosmsg::sensor_msgs::JointState
     //        * Add a map string --> indeces
     // Fill the buffer of joints positions
     const iDynTree::Model& model = m_kinDynComp.model();
+    iDynTree::JointIndex jntIndex;
     for (size_t i=0; i < v.name.size(); i++)
     {
-        iDynTree::JointIndex jntIndex = model.getJointIndex(v.name[i]);
+        for (size_t j=0; j < model.getNrOfDOFs(); j++)
```

Comment:
This is working with models for which fixed joints are added **after** all the joints with non-zero degrees of freedom, such as the one created by by our URDF parser, but it can fail for general iDynTree::Model for which for which it can happend that a non-zero dof joint has a JointIndex that is greater then the `getNrOfDoFs`.